### ISSUE TITLE
Create output file with append=True

### ISSUE THREAD

**Issue by davidoskky** [openmm/openmm#4167]
Setting the parameter append=True to DCDReporter produces a crash if the file does not exist. I'd rather have it automatically create the file.

**Comment by peastman** [comment]
When you say a crash, I assume you mean an exception telling you the file doesn't exist?

I'm not sure about this.  If you tell it to append to an existing file, but there is no existing file, that could easily indicate a problem that the user really needs to be told about.  You aren't doing them any service by silently ignoring the problem.

**Comment by RaulPPelaez** [comment]
I agree with peastman. Furthermore you can just mimic this behavior with a single line, e.g:
```python
...
filename='traj.dcd'
open(filename, 'a').close()
simulation.reporters.append(DCDReporter(filename, 1000, append=True))
```

**Comment by swails** [comment]
> I'm not sure about this. If you tell it to append to an existing file, but there is no existing file, that could easily indicate a problem that the user really needs to be told about. You aren't doing them any service by silently ignoring the problem.

A counterargument to this is that the default C-like language behavior for file appending -- including Python itself (`open(filename, 'a')`) -- is to create the file if it does not exist, and write if it does.

I'm +1 on the recommended change here.  If you can use "append" to create-or-append, then you can code in a way that ensures you'll never lose data accidentally.  Otherwise you'll need to write some code like this:

```python
if os.path.exists("<filename>"):
    fh = open("<filename>", "a")
else:
    fh = open("<filename>", "w")
```

But what if we mess up by accident and swap the "a" and "w" (an easy mixup to do) -- we'll find ourselves accidentally overwriting the file if it exists.

Compare that to `fh = open('filename', 'a')` always doing the right thing.

It's hard to really argue this is surprising behavior when it's the default file behavior in most languages (including Python).

You could always add a new keyword `create=True` so that the "create-if-it-doesn't-exist" semantics are opt-in if you're worried about the file not existing being a symptom of an error worth reporting on.

On balance, I think the current behavior is more susceptible to accidents than the proposed behavior.
